internal/auth0/connection/expand.go

```diff
@@ -555,6 +559,21 @@ func expandConnectionOptionsAD(config cty.Value) (*management.ConnectionOptionsA
 		BruteForceProtection: value.Bool(config.GetAttr("brute_force_protection")),
 	}
 
+	if !d.IsNewResource() {
+		conn, err := api.Connection.Read(d.Id())
```

Comment:
We don't really need to pass the API here in all these places and force another request to fetch the current details. We can add all the extra properties as read-only (Computed), and simply update all the expands to fetch those details as well and pass them through, effectively skipping the API call.

Reply:
These values don't have value to the end-user, they're useful for facilitating the operation of the connection. Exposing them through just pollutes the schema and could confuse end-users.

Reply:
Alright, that makes sense. However the expand is now doing a bit too much, it would be nice to have these "supplements" in separate funcs within the main expand switch statement.